src/App.tsx

```diff
@@ -1,14 +1,43 @@
 import React from 'react';
-
 import './App.scss';
+import {
+  Post,
+  PreparedPost,
+  User,
+  PostComment,
+} from './appTypeDefs';
+
+import postsFromServer from './api/posts';
+import commentsFromServer from './api/comments';
+import usersFromServer from './api/users';
+import { PostList } from './components/PostList/PostList';
 
-// import posts from './api/posts';
-// import comments from './api/comments';
-// import users from './api/users';
+export function preparePosts(
+  posts: Post[],
+  postComments: PostComment[],
+  users: User[],
+): PreparedPost[] {
+  return posts.map(post => ({
+    ...post,
+    user: users.find(user => (
+      user.id === post.userId
+    )) || null || undefined,
+    postComments: postComments.filter(postComment => (
+      postComment.postId === post.id
+    )) || null || undefined,
+  }));
+}
+
+export const preparedPosts = preparePosts(
+  postsFromServer,
+  commentsFromServer,
+  usersFromServer,
+);
```

Comment:
I'm not sure about the helper function I don't think you really need it since you only use it once. I think you could've assigned the body of the return statement to `preparedPosts`

Reply:
Okay yah you are right. Have fixed it. Thanx